tests/test_integrals.py

```diff
@@ -692,28 +695,474 @@ def test_vanilla_params(self):
+            B0n=-surf.compute("n_rho", grid=src_grid)["n_rho"][:, 2],
+            use_dft=use_dft,
+            warn_dft=False,
+            warn_fft=False,
+        )
+
+        data = vac.compute_Phi(chunk_size)
+        Phi = Transform(vac.evl_grid, vac.basis).transform(data["Phi"]["Phi_mn"])
+        Z = data["evl"]["Z"]
+        np.testing.assert_allclose(np.ptp(Z - Phi), 0, atol=atol)
+
+        data = vac.compute_vacuum_field(chunk_size)["evl"].copy()
+        data = surf.compute("n_rho", grid=vac.evl_grid, data=data)
+        B0n = -data["n_rho"][:, 2]
+        dPhi_dn = dot(data["grad(Phi)"], data["n_rho"])
+        np.testing.assert_allclose(B0n + dPhi_dn, 0, atol=atol)
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("chunk_size", [25])
+    def test_harmonic_exterior(self, chunk_size):
+        """Test that Laplace solution recovers expected analytic result.
+
+        Define harmonic map Φ: ρ,θ,ζ ↦ G(ρ,θ,ζ).
+        Choose b.c. 𝐁₀⋅𝐧 = -∇G⋅𝐧 and test that ‖ Φ − G ‖_∞ → 0.
+        """
+        atol = 1e-3
+        # elliptic cross-section with torsion
+        surf = FourierRZToroidalSurface(
+            R_lmn=[100, 10, 1],
+            Z_lmn=[-20, -1],
+            modes_R=[[0, 0], [1, 0], [0, 1]],
+            modes_Z=[[-1, 0], [0, -1]],
+        )
+        src_grid = LinearGrid(M=30, N=50, NFP=surf.NFP)
+        Phi_grid = LinearGrid(M=30, N=50, NFP=surf.NFP)
+
+        src_data = surf.compute(["x", "n_rho"], grid=src_grid, basis="xyz")
+        # place green's function origin inside torus
+        # so that singularity is outside of the domain of the
+        # exterior problem
+        y0 = jnp.array([100, 0, 0])
+
+        def grad_G(x):
+            # ∇G(x) = -∇_y G(x-y)
+            y = y0
+            return -_grad_G(x - y)
+
+        vac = VacuumSolver(
+            surface=surf,
+            evl_grid=Phi_grid,
+            src_grid=src_grid,
+            Phi_grid=Phi_grid,
+            Phi_M=20,
+            Phi_N=20,
+            exterior=True,
+            chunk_size=chunk_size,
+            B0n=-dot(grad_G(src_data["x"]), src_data["n_rho"]),
+            use_dft=False,
+            warn_dft=False,
+            warn_fft=False,
+            sym=False,
+        )
+
+        evl_data = surf.compute(["x", "n_rho", "phi"], grid=vac.evl_grid, basis="xyz")
+
+        data = vac.compute_Phi(chunk_size)
+        Phi = Transform(vac.evl_grid, vac.basis).transform(data["Phi"]["Phi_mn"])
+        G = np.reciprocal(_1_over_G(evl_data["x"] - y0))
+        np.testing.assert_allclose(np.ptp(G - Phi), 0, atol=atol)
+
+        data = vac.compute_vacuum_field(chunk_size)["evl"]
+        grad_Phi = rpz2xyz_vec(data["grad(Phi)"], phi=evl_data["phi"])
+        np.testing.assert_allclose(
+            dot(grad_G(evl_data["x"]) - grad_Phi, evl_data["n_rho"]),
+            0,
+            atol=atol,
+        )
+
+    @pytest.mark.unit
+    def test_secular_splitting(self, chunk_size=100, I=0, Y=12):  # noqa: E741
```

Comment:
@dpanici 

I am going to table this approach for now. When we want to add the effect of secular terms we can just compute the `B0` field that generates that secular term and use that method of splitting the potential as described in my report. That is the standard way of doing it and our functionality to that is already working and tested.